templates/repo/issue/branch_selector_field.tmpl

```diff
@@ -1,5 +1,11 @@
 {{if and (not .Issue.IsPull) (not .PageIsComparePull)}}
 <input id="ref_selector" name="ref" type="hidden" value="{{.Issue.Ref}}">
+<input id="editing_mode" name="edit_mode" type="hidden" value="{{.IsIssueWriter}}">
```

Comment:
```suggestion
<input id="editing_mode" name="edit_mode" type="hidden" value="{{(or .IsIssueWriter .IsIssuePoster)}}">
```
This will make it consistent with the backend. However, I'm not sure what the permission requirements should be for changing the issue branch; write issue permissions, merge permissions?

Reply:
My idea was that only write access/assignee should be allowed to change branch. In my opinion, creator should not be allowed to change it unless it fits the first criteria (or even assign a branch to an issue in the first place). I would base this on the assumption that an issue can be created by anyone, but only those who can solve it should be able to link it to a branch.

But then I was confused about IsIssueWriter here. Comment in the code says:

> returns true if isPull is true and user could write to pull requests and returns true if isPull is false and user could write to issues

The way I interpreted is that if it can create issues, which any user on a public repo can, it could change a branch. In practice, seems that it needs write access which is what I would prefer here.

Reply:
It's confusing indeed, but it's not what it seems:

**Issue read permissions**: Can read issues, can comment on any issue, can create issues, can delete/modify/close own issues.
**Issue write permissions**: All of the above plus can delete/modify/close other people's issues.

Hope this helps. 😄

Reply:
It does indeed, thanks ^^